frontend/pages/hosts/details/MacSettingsModal/MacSettingsTable/MacSettingsTableConfig.tsx

```diff
@@ -0,0 +1,109 @@
+import LinkCell from "components/TableContainer/DataTable/LinkCell";
+import TextCell from "components/TableContainer/DataTable/TextCell";
+import React from "react";
+import { IMacSetting, IMacSettings } from "interfaces/mdm";
+import MacSettingsIndicator from "../../MacSettingsIndicator";
+
+interface IHeaderProps {
+  column: {
+    title: string;
+    isSortedDesc: boolean;
+  };
+}
+
+interface ICellProps {
+  cell: {
+    value: string;
+  };
+  row: {
+    original: IMacSetting;
+  };
+}
+
+interface IDataColumn {
+  Header: ((props: IHeaderProps) => JSX.Element) | string;
+  Cell: (props: ICellProps) => JSX.Element;
+  id?: string;
+  title?: string;
+  accessor?: string;
+  disableHidden?: boolean;
+  disableSortBy?: boolean;
+  sortType?: string;
+}
+
+const settingStatusOptions = {
+  "Action required (pending)": {
+    iconName: "pending",
+    tooltipText: "Follow Disk encryption instructions on your My device page.",
+  },
+  Applied: {
+    iconName: "success",
+    tooltipText: "Disk encryption on and disk encryption key stored in Fleet.",
+  },
+  "Enforcing (pending)": {
+    iconName: "pending",
+    tooltipText: "Setting will be enforced when the host comes online.",
+  },
+  "Removing enforcement (pending)": {
+    iconName: "pending",
+    tooltipText: "Enforcement will be removed when the host comes online.",
+  },
+  Failed: { iconName: "error", tooltipText: null },
+} as const;
+
+const generateTableHeaders = (): IDataColumn[] => {
```

Comment:
Since these are basically static (i.e. there are no parameters to this function), you can eliminate the function and just export the tableHeaders constant.